components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/RequestCountResource.java

```diff
@@ -0,0 +1,135 @@
+            if (!StringUtils.isEmpty(yearParameter) && !StringUtils.isEmpty(monthParameter)) {
+                return takeRequestCountOfTheMonth(synCtx, Integer.parseInt(yearParameter),
+                        Integer.parseInt(monthParameter));
+            }
+        } else if (StringUtils.isEmpty(yearParameter) && StringUtils.isEmpty(monthParameter)) {
+            Date date = new Date();
+            LocalDate localDate = date.toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
+            return takeRequestCountOfTheMonth(synCtx, localDate.getYear(), localDate.getMonthValue());
+        }
+        JSONObject response = Utils.createJsonError("Input parameters are not valid", axisCtx, BAD_REQUEST);
+        Utils.setJsonPayLoad(axisCtx, response);
+        return true;
+    }
+
+    // Take aggregated request count for a given month.
+    private boolean takeRequestCountOfTheMonth(MessageContext synCtx, int year, int month) {
+
+        JSONObject response;
+        org.apache.axis2.context.MessageContext axisCtx = ((Axis2MessageContext) synCtx).getAxis2MessageContext();
+        axisCtx.setProperty(org.apache.axis2.Constants.Configuration.MESSAGE_TYPE, "application/json");
+        axisCtx.setProperty(Constants.Configuration.CONTENT_TYPE, "application/json");
+
+        TransactionStore transactionStore = DataHolder.getInstance().getTransactionStore();
+        if (transactionStore != null) {
+            // Get request count of current Month
+            long requestCount;
+            try {
+                requestCount = transactionStore.getTransactionCountOfMonth(year, month);
+            } catch (TransactionCounterException e) {
+                response =
+                        Utils.createJsonError("Error occurred while retrieving data from the database ", e, axisCtx,
+                                INTERNAL_SERVER_ERROR);
+                Utils.setJsonPayLoad(axisCtx, response);
+                return true;
+            }
+            if (requestCount != -1L) {
+                response = new JSONObject(
+                        "{\"Year\" : " + year + ", \"Month\" : " + month + ", \"RequestCount\" : " + requestCount +
+                                "}");
+            } else {
+                response = new JSONObject("{\"error\" : \"Did not find stats for " + year + "/" + month + "\"}");
+            }
+        } else {
+            response =
+                    Utils.createJsonError("TransactionStore is not initialized", axisCtx, FORBIDDEN);
+        }
+        Utils.setJsonPayLoad(axisCtx, response);
+        return true;
+    }
+
+    /**
+     * Try to parse input string to int.
+     *
+     * @param someText input string.
+     * @return parsed integer, null if failed.
+     */
+    private static Integer tryParseInt(String someText) {
+
+        try {
+            return Integer.parseInt(someText);
+        } catch (NumberFormatException ex) {
+            return null;
```

Comment:
Shall we print an error log here so that we can refer in server side.

Reply:
+1. fixed.